src/aks-preview/azext_aks_preview/aks_app_commands/aksappcommands.py

```diff
@@ -0,0 +1,150 @@
+    return None
+
+
+# Returns a list of potential draftV2 binary paths
+def _get_potential_paths() -> List[str]:
+    result = [str(Path.home()) + '/' + '.aksapp']
+    paths = os.environ['PATH'].split(':')
+
+    for path in paths:
+        if 'draftv2' in path:
+            result.append(path)
+    return result
+
+
+def _build_arguments(app_name: str,
+                     language: str,
+                     create_config: str,
+                     dockerfile_only: str,
+                     deployment_only: str) -> List[str]:
+    options = {
+        'app-name': app_name,
+        'language': language,
+        'create-config': create_config,
+        'dockerfile-only': dockerfile_only,
+        'deployment-only': deployment_only
+    }
+    args_list = []
+    for arg, val in options.items():
+        if val:
+            args_list.append(f'--{arg}={val}')
+    return args_list
+
+
+def _run(binary_path: str, deployment_path: str, arguments: List[str]) -> bool:
+    if binary_path is None:
+        raise ValueError('The given Binary path was null or empty')
+
+    print('Running DraftV2 Binary ...')
+    cmd = [binary_path, 'create', deployment_path] + arguments
+    process = subprocess.Popen(cmd)
+    exit_code = process.wait()
+    return exit_code == 0
+
+
+def _cmd_finish():
+    # Clean up logic can go here if needed
+    pass
+
+
+def _download_binary() -> str:
+    # prompt user to download binary. If users says no, we error out and tell them that this requires the binary
+    from knack.prompting import prompt_y_n
+    msg = 'The required binary was not found. Would you like us to download the required binary for you?'
+
+    if not prompt_y_n(msg, default='n'):
+        raise ValueError('`az aks app` requires the missing dependency')
+
+    print('Attempting to download dependency...')
+
+    operating_system = platform.system()
+    draftv2_release_version = 'v0.0.5'
+    filename = 'draftv2-' + operating_system.lower() + '-amd64'
+    url = 'https://github.com/Azure/aks-app/releases/download/' + draftv2_release_version + '/' + filename
+    headers = {'Accept': 'application/octet-stream'}
+
+    dir_name = '.aksapp'
+    # Downloading the file by sending the request to the URL
+    req = requests.get(url, headers=headers)
+    binary_path = str(Path.home()) + '/' + dir_name
+
+    # Directory
+    if os.path.exists(binary_path) is False:
+        os.chdir(str(Path.home()))
+        os.mkdir(dir_name)
+        print(f'Directory {dir_name} was created inside of your HOME directory')
+
+    if req.ok:
+        # Split URL to get the file name 'draftv2-darwin-amd64'
+        os.chdir(binary_path)
+        # Writing the file to the local file system
+        with open(filename, 'wb') as output_file:
+            output_file.write(req.content)
+        print('Download of DraftV2 binary was successful with a status code: ' + str(req.status_code))
+        os.chmod(binary_path + '/' + filename, 0o777)
```

Comment:
🐛 🪲`Security issue` - This could cause a security issue, because whatever the binary is we are providing `read, write and execute` once downloaded we could do, `0o655` or whatever is preferred permissioning model.